### PR TITLE
feat: support parsing space for deletion #3418

### LINKED ISSUE #3418: feature(cli): support parsing space for deletion
### Feature request

Right now we don't support parsing space

```python
bentoml models delete model1 model2 ...
```
will fail.

Note that we can already do

```python
bentoml models delete model1,model2
```

### Motivation

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by OmkarMozar**
## What does this PR address?

This PR adds support for parsing space in delete command for both `benotml delete` and `bentoml models delete`
Instead of returning value argument as a STRING , It is removed to take default one. Also added nrgs=-1 to have functionality of multiple targets.
Rest of the code is just parsing.

**TEST SNAPSHOTS:**
Issue reproduce (bentoml models delete):
```
(base) vrushali@Vrushalis-MacBook-Air simple_pickable_model % bentoml models list
 Tag                               Module                   Size      Creation Time       
 my_python_model:wurpysaau6io4js4  bentoml.picklable_model  1.01 KiB  2023-06-01 23:41:19 
 iris_clf:i7wa4yqau6dnajs4         bentoml.sklearn          5.71 KiB  2023-06-01 23:38:16 
 linear_reg:l7l4uqh6a6mvejs4       bentoml.sklearn          940.00 B  2023-05-29 15:28:34 
(base) vrushali@Vrushalis-MacBook-Air simple_pickable_model % bentoml models delete my_python_model:wurpysaau6io4js4 iris_clf:i7wa4yqau6dnajs4 linear_reg:l7l4uqh6a6mvejs4
Usage: bentoml models delete [OPTIONS] DELETE_TARGETS
Try 'bentoml models delete -h' for help.

Error: Got unexpected extra arguments (iris_clf:i7wa4yqau6dnajs4 linear_reg:l7l4uqh6a6mvejs4)
```
After Issue Fixes (bentoml models delete):
```
(base) vrushali@Vrushalis-MacBook-Air BentoML % bentoml models delete my_python_model:wurpysaau6io4js4 iris_clf:i7wa4yqau6dnajs4 linear_reg:l7l4uqh6a6mvejs4
delete model my_python_model:wurpysaau6io4js4? [y/N]: y
Model(tag="my_python_model:wurpysaau6io4js4") deleted.
delete model iris_clf:i7wa4yqau6dnajs4? [y/N]: y
Model(tag="iris_clf:i7wa4yqau6dnajs4") deleted.
delete model linear_reg:l7l4uqh6a6mvejs4? [y/N]: y
Model(tag="linear_reg:l7l4uqh6a6mvejs4") deleted.
(base) vrushali@Vrushalis-MacBook-Air BentoML % 

**USING COMMA separated :**

(base) vrushali@Vrushalis-MacBook-Air simple_pickable_model % bentoml models delete my_python_model:obhsbaabggj5qjs4,iris_clf:mqhgz2qbggi6ijs4,linear_reg:eugaadabgggnujs4
delete model my_python_model:obhsbaabggj5qjs4? [y/N]: y
Model(tag="my_python_model:obhsbaabggj5qjs4") deleted.
delete model iris_clf:mqhgz2qbggi6ijs4? [y/N]: y
Model(tag="iris_clf:mqhgz2qbggi6ijs4") deleted.
delete model linear_reg:eugaadabgggnujs4? [y/N]: y
Model(tag="linear_reg:eugaadabgggnujs4") deleted.
(base) vrushali@Vrushalis-MacBook-Air simple_pickable_model % 
```
Issue reproduce (bentoml delete):
```
(base) vrushali@Vrushalis-MacBook-Air BentoML % bentoml list                                                                                                         
 Tag                                 Size       Creation Time       
 linear_regression:pm6ixwqbgk57mjs4  14.92 KiB  2023-06-02 16:14:42 
 iris_classifier:ljzurzabgkx24js4    25.10 KiB  2023-06-02 16:14:02 
 simple_square_svc:enxfuwqbgkkrqjs4  15.10 KiB  2023-06-02 16:12:19 
(base) vrushali@Vrushalis-MacBook-Air BentoML % bentoml delete linear_regression:pm6ixwqbgk57mjs4 iris_classifier:ljzurzabgkx24js4 simple_square_svc:enxfuwqbgkkrqjs4
Usage: bentoml delete [OPTIONS] DELETE_TARGETS
Try 'bentoml delete -h' for help.

Error: Got unexpected extra arguments (iris_classifier:ljzurzabgkx24js4 simple_square_svc:enxfuwqbgkkrqjs4)
```
After Issue Fixes (bentoml delete):
```
(base) vrushali@Vrushalis-MacBook-Air BentoML % bentoml list                                                                                                         
 Tag                                 Size       Creation Time       
 linear_regression:pm6ixwqbgk57mjs4  14.92 KiB  2023-06-02 16:14:42 
 iris_classifier:ljzurzabgkx24js4    25.10 KiB  2023-06-02 16:14:02 
 simple_square_svc:enxfuwqbgkkrqjs4  15.10 KiB  2023-06-02 16:12:19 
(base) vrushali@Vrushalis-MacBook-Air BentoML % bentoml delete linear_regression:pm6ixwqbgk57mjs4 iris_classifier:ljzurzabgkx24js4 simple_square_svc:enxfuwqbgkkrqjs4
delete bento linear_regression:pm6ixwqbgk57mjs4? [y/N]: y
Bento(tag="linear_regression:pm6ixwqbgk57mjs4") deleted.
delete bento iris_classifier:ljzurzabgkx24js4? [y/N]: y
Bento(tag="iris_classifier:ljzurzabgkx24js4") deleted.
delete bento simple_square_svc:enxfuwqbgkkrqjs4? [y/N]: y
Bento(tag="simple_square_svc:enxfuwqbgkkrqjs4") deleted.

**COMMA separated:**

(base) vrushali@Vrushalis-MacBook-Air quickstart % bentoml list                                                                                                         
 Tag                                 Size       Creation Time       
 iris_classifier:pcb53labgozryjs4    25.10 KiB  2023-06-02 16:21:49 
 simple_square_svc:nrtoqsabgo3t4js4  15.10 KiB  2023-06-02 16:21:28 
 linear_regression:miggyeabgoh7mjs4  14.92 KiB  2023-06-02 16:21:09 
(base) vrushali@Vrushalis-MacBook-Air quickstart % bentoml delete iris_classifier:pcb53labgozryjs4,simple_square_svc:nrtoqsabgo3t4js4,linear_regression:miggyeabgoh7mjs4
delete bento iris_classifier:pcb53labgozryjs4? [y/N]: y
Bento(tag="iris_classifier:pcb53labgozryjs4") deleted.
delete bento simple_square_svc:nrtoqsabgo3t4js4? [y/N]: y
Bento(tag="simple_square_svc:nrtoqsabgo3t4js4") deleted.
delete bento linear_regression:miggyeabgoh7mjs4? [y/N]: y
Bento(tag="linear_regression:miggyeabgoh7mjs4") deleted.
(base) vrushali@Vrushalis-MacBook-Air quickstart % bentoml list                                                                                                         
 Tag  Size  Creation Time 
(base) vrushali@Vrushalis-MacBook-Air quickstart % 

```

Fixes #3418

## Before submitting:

- [X] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [X] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [X] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [X] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [X] Did you write tests to cover your changes?
- IF applicable I will do that in follow up issue.

### CODE REVIEW

File: src/bentoml_cli/bentos.py [hunk 1]
```diff
@@ -164,6 +168,7 @@ def delete(delete_targets: list[str], yes: bool) -> None:  # type: ignore (not a
             * Delete single bento bundle by "name:version", e.g: `bentoml delete IrisClassifier:v1`
             * Bulk delete all bento bundles with a specific name, e.g.: `bentoml delete IrisClassifier`
             * Bulk delete multiple bento bundles by name and version, separated by ",", e.g.: `benotml delete Irisclassifier:v1,MyPredictService:v2`
+            * Bulk delete multiple bento bundles by name and version, separated by " ", e.g.: `benotml delete Irisclassifier:v1 MyPredictService:v2`
```
Comment (sauyon) [inline src/bentoml_cli/bentos.py:171]: Here too!

**Review by sauyon — COMMENTED** [review]
LGTM! Just one (or two) minor things.

File: src/bentoml_cli/models.py [hunk 2]
```diff
@@ -155,6 +159,7 @@ def delete(delete_targets: str, yes: bool) -> None:  # type: ignore (not accesse
             * Delete single model by "name:version", e.g: `bentoml models delete iris_clf:v1`
             * Bulk delete all models with a specific name, e.g.: `bentoml models delete iris_clf`
             * Bulk delete multiple models by name and version, separated by ",", e.g.: `benotml models delete iris_clf:v1,iris_clf:v2`
+            * Bulk delete multiple models by name and version, separated by " ", e.g.: `benotml models delete iris_clf:v1 iris_clf:v2`
```
Comment (aarnphm) [inline src/bentoml_cli/models.py:162]:
```suggestion
            * Bulk delete multiple models by name and version, separated by ",", e.g.: `bentoml models delete iris_clf:v1,iris_clf:v2`
            * Bulk delete multiple models by name and version, separated by " ", e.g.: `bentoml models delete iris_clf:v1 iris_clf:v2`
```

File: src/bentoml_cli/bentos.py (hunk 1 above)
Comment (aarnphm) [inline src/bentoml_cli/bentos.py:171]:
```suggestion
            * Bulk delete multiple bento bundles by name and version, separated by ",", e.g.: `bentoml delete Irisclassifier:v1,MyPredictService:v2`
            * Bulk delete multiple bento bundles by name and version, separated by " ", e.g.: `bentoml delete Irisclassifier:v1 MyPredictService:v2`
```

**Review by aarnphm — APPROVED** [review]

**Review by aarnphm — APPROVED** [review]

**Comment by aarnphm** [comment]
Thanks again for this LGTM.

**Pull request merged**